### PR TITLE
refactor(client): literal

### PR DESCRIPTION
**Pull request by javadkh2**
refactor literal in order to create correct code if users pass object to a function

### CODE REVIEW

**Comment by javadkh2** [comment]
I need to fix the type issue

File: packages/libs/pactjs-generator/src/contract/generation/generator.ts [hunk 1]
```diff
@@ -139,7 +139,10 @@ export function generateDts(
       .join(EOL.repeat(2)) || '';
 
   const dts = `
-import type { IPactDecimal, IPactInt, PactReference, ICap } from '@kadena/types';
+import type { IPactDecimal, IPactInt, ICap } from '@kadena/types';
+import type { Literal } from '@kadena/client';
+
+type PactReference = Literal | (() => string);
```
Comment (alber70g) [inline packages/libs/pactjs-generator/src/contract/generation/generator.ts:145]: Can we move this one to `@kadena/client` and import it from there?

File: packages/libs/pactjs-generator/src/contract/generation/generator.ts [hunk 2]
```diff
@@ -139,7 +139,8 @@ export function generateDts(
       .join(EOL.repeat(2)) || '';
 
   const dts = `
-import type { IPactDecimal, IPactInt, PactReference, ICap } from '@kadena/types';
+import type { IPactDecimal, IPactInt, ICap } from '@kadena/types';
+import type { PactReference } from '@kadena/client';
```
Comment (alber70g) [inline packages/libs/pactjs-generator/src/contract/generation/generator.ts:143]: This should be the other way around as imports are sorted alphabetically.

**Review by alber70g — APPROVED** [review]

File: common/changes/@kadena/types/fix-literal_2023-08-16-09-24.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@kadena/types",
+      "comment": "depricated the PactReference infavor of Literal class",
```
[the remark this answers is not in the document]
Comment (javadkh2) [reply common/changes/@kadena/types/fix-literal_2023-08-16-09-24.json:5]: Actually I have removed the PactReference, I will update the comment

**Pull request merged**